File: models/experiment.go
package models

// Experiment is model for an experiment.
type Experiment struct {
	ID            int           `json:"experiment_id"`
	Cell          Cell          `json:"cell_line"`
	Tissue        Tissue        `json:"tissue"`
	Drug          Drug          `json:"drug"`
	Dataset       Dataset       `json:"dataset"`
	DoseResponses DoseResponses `json:"dose_responses,omitempty"`
}

// Experiments is a collection of Experiment.
type Experiments []Experiment

File: front-end/ruby-on-rails/app/models/experiment.rb
class Experiment < ApplicationRecord
	has_many :dose_responses
	has_one :profile
	belongs_to :cell
	belongs_to :drug
	belongs_to :dataset
	belongs_to :tissue
	has_many :source_cell_names, through: :cell
	has_many :source_drug_names, through: :drug
	has_many :drug_targets, through: :drug
	has_many :targets, through: :drug_targets
end
